tests/juniper_mx/juniper_mx_protocol_test.py

```diff
@@ -413,6 +461,59 @@ def test_trunk_mode(self):
         }))
         assert_that(result.xpath("data/configuration/bridge-domains/domain"), has_length(0))
 
+    def test_trunk_mode_delete_all_vlan(self):
+        self.edit({
+            "bridge-domains": [
+                {"domain": [
+                    {"name": "VLAN2995"},
+                    {"vlan-id": "2995"}]},
+                {"domain": [
+                    {"name": "VLAN2996"},
+                    {"vlan-id": "2996"}]},
+                {"domain": [
+                    {"name": "VLAN2997"},
+                    {"vlan-id": "2997"}]},
+            ],
+            "interfaces": {
+                "interface": [
+                    {"name": "xe-0/0/3"},
+                    {"native-vlan-id": "2996"},
+                    {"unit": [
+                        {"name": "0"},
+                        {"family": {
+                            "bridge": [
+                                {"interface-mode": "trunk"},
+                                {"vlan-id-list": "2995"},
+                                {"vlan-id-list": "2997"}
+                            ]}}]}]}})
+        self.nc.commit()
+
+        self.edit({
+            "interfaces": {
+                "interface": [
+                    {"name": "xe-0/0/3"},
+                    {"unit": [
+                        {"name": "0"},
+                        {"family": {
+                            "bridge": [{
+                                "vlan-id-list": {XML_ATTRIBUTES: {"operation": "delete"}}},
+                                {"vlan-id-list": "2996"}]
+                                }}]}]}})
+        self.nc.commit()
+        result = self.nc.get_config(source="running", filter=dict_2_etree({"filter": {
+            "configuration": {"interfaces": {"interface": {"name": "xe-0/0/3"}}}}
+        }))
+        int003 = result.xpath("data/configuration/interfaces/interface")[0]
+        assert_that(int003.xpath("unit/family/bridge/vlan-id-list"), has_length(1))
+        assert_that(int003.xpath("unit/family/bridge/vlan-id-list")[0].text, equal_to("2996"))
+
+        self.cleanup(vlan("VLAN2995"), vlan("VLAN2996"), vlan("VLAN2997"),
+                     reset_interface("xe-0/0/3"))
+        result = self.nc.get_config(source="running", filter=dict_2_etree({"filter": {
```

Comment:
nit: this verification seems already done by the test above

Reply:
It's vlan-id vs vlan-id-list removal. So, not quite the same code path.

Reply:
Unless you are talking about the cleanup assertion?
Yeah, that could be removed.